v3.0.0-rc1: mimxrt685_evk_cm33: undefined reference to `SystemCoreClock' for latency_measure benchmark

**Describe the bug**
undefined reference to `SystemCoreClock'

Please also mention any information which could help others to understand
the problem you're facing:
mimxrt685_evk_cm33
tests/benchmarks/latency_measure build failure


**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=mimxrt685_evk_cm33
3. make
4. See error

**Expected behavior**
build failure

**Impact**
kernel

**Logs and console output**

```

[163/173] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/ubuntu/zephyr_project/zephyr_rt685/zephyr/tests/benchmarks/latency_measure/build/zephyr/zephyr_pre0.map
: && ccache /home/ubuntu/zephyr-sdk/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc   zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/home/ubuntu/zephyr_project/zephyr_rt685/zephyr/tests/benchmarks/latency_measure/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/cmse/libarch__arm__core__aarch32__cortex_m__cmse.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/tz/libarch__arm__core__aarch32__cortex_m__tz.a  zephyr/arch/arch/arm/core/aarch32/mpu/libarch__arm__core__aarch32__mpu.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/posix/liblib__posix.a  zephyr/soc/arm/common/cortex_m/libsoc__arm__common__cortex_m.a  zephyr/boards/arm/mimxrt685_evk/libboards__arm__mimxrt685_evk.a  zephyr/subsys/timing/libsubsys__timing.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  modules/hal_nxp/hal_nxp/lib..__modules__hal__nxp.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/home/ubuntu/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/thumb/v8-m.main/nofp"  -L/home/ubuntu/zephyr_project/zephyr_rt685/zephyr/tests/benchmarks/latency_measure/build/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -no-pie  -mcpu=cortex-m33  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn && cd /home/ubuntu/zephyr_project/zephyr_rt685/zephyr/tests/benchmarks/latency_measure/build/zephyr && /home/ubuntu/.local/lib/python3.8/site-packages/cmake/data/bin/cmake -E echo
/home/ubuntu/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a(timing.c.obj): in function `z_arm_dwt_freq_get':
/home/ubuntu/zephyr_project/zephyr_rt685/zephyr/arch/arm/core/aarch32/cortex_m/timing.c:36: undefined reference to `SystemCoreClockUpdate'
/home/ubuntu/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a(timing.c.obj): in function `arch_timing_freq_get_mhz':
/home/ubuntu/zephyr_project/zephyr_rt685/zephyr/arch/arm/core/aarch32/cortex_m/timing.c:126: undefined reference to `SystemCoreClock'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /home/ubuntu/.local/bin/cmake --build /home/ubuntu/zephyr_project/zephyr_rt685/zephyr/tests/benchmarks/latency_measure/build

```

OS: (e.g. Linux, , )
Toolchain (e.g Zephyr SDK, ...): 0.13.0
Commit SHA or Version used: v3.0.0-Rc1

--- a/west.yml
+++ b/west.yml
@@ -98,7 +98,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: aa294fff4e1c82fec0f4e2ab71362e5fb65485cc
+      revision: 9629f15fd80169cc9669e3fbb1260467c3aeb175
       path: modules/hal/nxp
       groups:
         - hal

